### ISSUE TITLE
Caja and Nautilus throws "This destination is read-only" error when performing file operations.

### ISSUE THREAD

**Issue by zavocc** [termux/proot#294]
**Problem description**
When using `caja` (MATE's file manager) or `nautilus` (GNOME files app), performing file operations such as creating new folder or file, moving or copying files, causes an error "This destination is read only" despite directories are read-writable (home directory in my case). However, other file managers like `thunar` works correctly.

### Video attachment showing error when copying/moving, creating new directories in Caja
[error](https://github.com/termux/proot/assets/119603393/06415c25-9097-436f-b001-8f9af80e54bc)

### Screenshots, both Caja and Nautilus attempting to copy nohup.out file from Desktop to home directory through drag n drop
![error](https://github.com/termux/proot/assets/119603393/e09520cc-0320-4813-a2ff-958231e9af6a)
![error](https://github.com/termux/proot/assets/119603393/f3734c17-6760-4da3-b210-52a5620daddc)

**Steps to reproduce**
1. Install `caja` and/or `nautilus`
2. Perform simple file operations (e.g. create a new folder/file, move/copy files) whether if its drag n drop or manual

**Expected behavior**
It should work.

**Additional information**
Additionally, there are no terminal logs related to this problem when I launched either with a Terminal and perform file operations, it doesn't show relevant logs. Though everytime file IO is done causing an error, it prints `Could not inhibit power management; org.gnome.SessionManager` or `No such method "Inhibit"` warnings though it's kinda unrelated.

Android 11, aarch64

**Comment by fish4terrisa-MSDSM** [comment]
Also got that on my Android 10,aarch64 with Arch Linux in proot. That problem really lasted for a long period of time. (maybe from the early of 2023 it already existed)
And also, it will also fail if you want to copy files to desktop, with the same error.

**Comment by zavocc** [comment]
this actually happened 5 years ago since I use these but nobody had bothered reporting it

**Comment by Biswa96** [comment]
The image and video links are not visible. Those probably need `![image][link]` markdown syntax.

**Comment by fish4terrisa-MSDSM** [comment]
> Interestingly, when using clipboard to cut or copy files into another writable location (e.g. Documents folder), the paste option in the caja context menu was grayed out.

That's becauce caja detected that it doesn't have premission to copy to there(althrough it actually have). Maybe someone can find out the method caja used to detect the premission.(Maybe caja is using an old syscall or other methods which are not supported by proot...)

**Comment by michalbednarski** [comment]
Looks like this is checked through inspecting `/proc/self/mountinfo`

<code>9974 9973 254:10 / / <ins><strong>ro</strong></ins>,noatime master:1 - ext4 /dev/block/dm-10 ro,seclabel</code>

In this case however, you cannot use `--bind` to override because `/proc/self` is symlink and resolution happens before applying bind mounts, I'll need to see how to fix this nicely

Repro (by using Gio which is used by Caja, Gio uses libmount)

```c
// apt install libglib2.0-dev
// gcc $(pkg-config --cflags gio-2.0) example.c $(pkg-config --libs gio-2.0)
#include <stdio.h>
#include <gio/gio.h>

int main() {
        GFile *dest;
        GFileInfo *fsinfo;
        dest = g_file_new_for_path("/home");
        fsinfo = g_file_query_filesystem_info (dest,
                        G_FILE_ATTRIBUTE_FILESYSTEM_READONLY,
                        0,
                        NULL);

        gboolean ret = g_file_info_get_attribute_boolean (fsinfo, G_FILE_ATTRIBUTE_FILESYSTEM_READONLY);
        printf("readonly=%d\n", ret);
}
```

**Comment by michalbednarski** [comment]
The issue should be fixed after `pkg up` to `proot 5.1.107-63`
